lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -2816,6 +2816,26 @@ class StoringDiagnosticConsumer : public swift::DiagnosticConsumer {
         break;
       }
 
+      // Swift may insert note diagnostics after an error diagnostic with fixits
+      // related to that error. Check if the lastest inserted diagnostic is an
```

Comment:
typo: latest

I'm curious: What does Clang differently that this isn't also a problem there, or is it also a problem there?

Reply:
And if it's also a problem there, should we move the code into the generic UserExpression.cpp?

Reply:
Interesting, looks like something similar is being done to deal with clang diagnostics. On `ClangExpressionParser`:

```
   226     case DiagnosticsEngine::Level::Note:
   227       m_manager->AppendMessageToDiagnostic(m_output);
   228       make_new_diagnostic = false;
   229
   230       // 'note:' diagnostics for errors and warnings can also contain Fix-Its.
   231       // We add these Fix-Its to the last error diagnostic to make sure
   232       // that we later have all Fix-Its related to an 'error' diagnostic when
   233       // we apply them to the user expression.
   234       auto *clang_diag = MaybeGetLastClangDiag();
   235       // If we don't have a previous diagnostic there is nothing to do.
   236       // If the previous diagnostic already has its own Fix-Its, assume that
   237       // the 'note:' Fix-It is just an alternative way to solve the issue and
   238       // ignore these Fix-Its.
   239       if (!clang_diag || clang_diag->HasFixIts())
   240         break;
   241       // Ignore all Fix-Its that are not associated with an error.
   242       if (clang_diag->GetSeverity() != eDiagnosticSeverityError)
   243         break;
   244       AddAllFixIts(clang_diag, Info);
   245       break;
```

Reply:
Good to know. I guess we might be able to refactor this into one layer above (assuming that both use the sam LLVM diagnostics engine) but it's not that much code, so I'll leave that decision up to you.